contracts/civic/src/civic.cpp

```diff
@@ -1,35 +1,87 @@
+#include <eosio/system.hpp>
+#include <eosio/time.hpp>
+
+
 #include <civic.hpp>
 
-// ACTION civic::hi(name from, string message) {
-//   require_auth(from);
-
-//   // Init the _message table
-//   messages_table _messages(get_self(), get_self().value);
-
-//   // Find the record from _messages table
-//   auto msg_itr = _messages.find(from.value);
-//   if (msg_itr == _messages.end()) {
-//     // Create a message record if it does not exist
-//     _messages.emplace(from, [&](auto& msg) {
-//       msg.user = from;
-//       msg.text = message;
-//     });
-//   } else {
-//     // Modify a message record if it exists
-//     _messages.modify(msg_itr, from, [&](auto& msg) {
-//       msg.text = message;
-//     });
-//   }
+ACTION civic::propcreate(name creator, string title, string description, string category, float budget, uint8_t type, uint8_t status, string location)
```

Comment:
status should be set automatically to Proposed, should not be an action argument